File: array/candy.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution {
 public:
  int minCandy(int N, vector<int> &ratings) {
    // code here
    vector<int> left(N), right(N);
    left[0] = 1;
    right[N - 1] = 1;
    for (int i = 1; i < N; ++i) {
      if (ratings[i] <= ratings[i - 1])
        left[i] = 1;
      else
        left[i] = left[i - 1] + 1;
    }
    for (int i = N - 2; i >= 0; --i) {
      if (ratings[i] <= ratings[i + 1])
        right[i] = 1;
      else
        right[i] = right[i + 1] + 1;
    }
    int ans = 0;
    for (int i = 0; i < N; ++i) {
      ans += max(left[i], right[i]);
    }
    return ans;
  }
};

//{ Driver Code Starts.

int main() {
  int t;
  cin >> t;
  while (t--) {
    int N;
    cin >> N;

    vector<int> ratings(N);
    for (int i = 0; i < N; i++) {
      cin >> ratings[i];
    }
    Solution obj;
    cout << obj.minCandy(N, ratings) << endl;
  }
  return 0;
}
// } Driver Code Ends

File: array/candy.py
# {
# Driver Code Starts
# Initial Template for Python 3

# } Driver Code Ends
# User function Template for python3


class Solution:
    def minCandy(self, N, ratings):
        # Code here
        left = [1] * N
        right = [1] * N
        for i in range(1, N):
            if ratings[i] > ratings[i - 1]:
                left[i] = left[i - 1] + 1

        for i in range(N - 2, -1, -1):
            if ratings[i] > ratings[i + 1]:
                right[i] = right[i + 1] + 1

        ans = 0
        for i in range(N):
            ans += max(left[i], right[i])

        return ans


# {
# Driver Code Starts.
if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        N = int(input())
        ratings = list(map(int, input().split()))
        ob = Solution()
        res = ob.minCandy(N, ratings)
        print(res)
# } Driver Code Ends
